CONTRIBUTING.md

```diff
@@ -500,6 +500,58 @@ it is acceptable to follow the convention in that module; though in general, it
 would be preferable to follow the motto of "leave the codebase in better shape
 than you found it."
 
+## Error Codes and Messages
+
```

Comment:
Since this is being added, how about we also document the general approach to error handling to help people who are not already aware of how `boost::outcome` or sum-type based error handling in general works? 

Here we can have links to `katana::Result` and `KATANA_ERROR`. And in the doc comments of those they can have examples of how they are used and links to `boost::outcome` for more details.